src/main/java/edu/ucla/library/prl/harvester/services/HarvestServiceUtils.java

```diff
@@ -0,0 +1,303 @@
+                        !stringifiedItemUrls.contains(value)) {
+                    if (dcElementsMap.containsKey(name)) {
+                        dcElementsMap.get(name).add(value);
+                    } else {
+                        // First instance of element
+                        final List<String> elements = new LinkedList<>();
+
+                        elements.add(value);
+                        dcElementsMap.put(name, elements);
+                    }
+                }
+            }
+
+            // Now that we know which values are item URLs and thumbnail URLs, we can avoid them
+            for (final Entry<String, List<String>> entry : dcElementsMap.entrySet()) {
+                doc.addField(entry.getKey() + "_keyword", entry.getValue());
+
+                switch (entry.getKey()) {
+                    case DC_DATE:
+                        final List<Integer> decades = getDecadesAscending(entry.getValue());
+
+                        if (!decades.isEmpty()) {
+                            doc.addField("decades", decades);
+                            doc.addField("sort_decade", decades.get(0));
+                        }
+                        break;
+                    case DC_TITLE:
+                        doc.addField("first_title", entry.getValue().get(0));
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            return doc;
+        });
+    }
+
+    /**
+     * @param aUrlList A list of URLs
+     * @return A list of the URLs in string form
+     */
+    private static List<String> unwrapUrls(final List<URL> aUrlList) {
+        final List<String> strings = new LinkedList<>();
+
+        for (final URL url : aUrlList) {
+            strings.add(url.toString());
+        }
+
+        return strings;
+    }
+
+    /**
+     * @param aDates A list of strings contained in a record's dc:date fields
+     * @return The list of decades to which all the dates in the record belong, sorted in ascending order
+     */
+    static List<Integer> getDecadesAscending(final List<String> aDates) {
+        // FIXME: this implementation is incorrect
+        return List.of(2000 + "FIXME".length());
+    }
+
+    /**
+     * Assigns a score to a URL based on how likely it is to be a record's canonical item URL, according to a simple
+     * heuristic.
+     * <p>
+     * A URL is considered more likely to be the canonical item URL if:
+     * <ul>
+     * <li>it contains the record identifier in its path part</li>
+     * <li>it has the same domain part as the OAI-PMH base URL</li>
+     * </ul>
+     *
+     * @param aURL The URL to score
+     * @param aRecordIdentifier The identifier of the record in which the URL was found
+     * @param aRepositoryURL The base URL from which the record was harvested
+     * @return The score
+     */
+    static int scoreURL(final URL aURL, final String aRecordIdentifier, final URL aRepositoryURL) {
```

Comment:
For reference, [the Python implementation](https://github.com/UCLALibrary/pacific-rim-library/blob/b276c409dbc7e03cafb26d362bb48121a234b886/pacific_rim_library/hyperlink_relevance_heuristic_sorter.py)